src/org/labkey/test/tests/list/ListTest.java

```diff
@@ -1812,6 +1812,210 @@ private void testTricky(String listName, boolean autoKey) throws IOException
+
+        //   with alternate keys
+        log("Clean out list before next import.");
+        importDataPage.cancel();
+        resetList();
+        log("With lookup validation on, import data into the second list using number-like lookup values expecting alternate keys but also accepting primary keys.");
+        bulkData = baseLookupFieldName + "\n" +
+                "1E2\n" + // valid alternate key looking like a number
+                keyNumber +"\n" + // valid alternate key same value as a primary key
+                ".123\n" + // valid alternate key looking like a float
+                "Lookup\n" + // valid alternate key that is a string
+                keyNumber + "\n" + // another copy
+                "102\n" + // valid number-like alternate key
+                actualValues.get(1).get(EscapeUtil.fieldKeyEncodePart(lookupKeyFieldName)) + "\n" // primary key value not matching an alternate key
+        ;
+        _listHelper.clickImportData()
+                .setText(bulkData)
+                .setImportLookupByAlternateKey(true)
+                .submit();
+        log("Verify the import succeeds and resolves the lookups appropriately.");
+        expectedData = List.of(
+                Map.of(baseLookupFieldKey, "1E2"),
+                Map.of(baseLookupFieldKey, keyNumber),
+                Map.of(baseLookupFieldKey, ".123"),
+                Map.of(baseLookupFieldKey, "Lookup"),
+                Map.of(baseLookupFieldKey, keyNumber),
+                Map.of(baseLookupFieldKey, "102"),
+                Map.of(baseLookupFieldKey, actualValues.get(1).get(EscapeUtil.fieldKeyEncodePart(lookupFieldName)))
+        );
+        validateDataRegionTableForTricky(expectedData);
+
+        bulkData = baseLookupFieldName + "\nInvalid";
+        importDataPage = _listHelper.clickImportData();
+        error = importDataPage
+                .setText(bulkData)
+                .setImportLookupByAlternateKey(true)
+                .submitExpectingError();
+        checker().withScreenshot().verifyEquals("Error message for invalid string alternate key not as expected", "Value 'Invalid' not found for field " + baseLookupFieldName + " in the current context.", error);
+
+        bulkData = baseLookupFieldName + "\n1234";
+        error = importDataPage
+                .setText(bulkData)
+                .setImportLookupByAlternateKey(true)
+                .submitExpectingError();
+        checker().withScreenshot().verifyEquals("Error message for invalid number-like alternate key not as expected", "Value '1234' was not present in lookup target 'lists." + lookupListName + "' for field '" + baseLookupFieldName + "'", error);
+
+    }
+
+    private void resetList()
+    {
+        DataRegionTable dataRegionTable = new DataRegionTable("query", getDriver());
+        dataRegionTable.checkAllOnPage();
+        dataRegionTable.deleteSelectedRows();
+    }
```

Comment:
Consider resetting list via api
```suggestion
    private void truncateList(String listName) throws IOException, CommandException
    {
        new QueryApiHelper(createDefaultConnection(), getProjectName(), "lists", listName).truncateTable();
    }
```

Reply:
Done.